sql-files/main.sql

```diff
@@ -107,12 +107,15 @@ CREATE TABLE IF NOT EXISTS `db_roulette` (
 --
 
 CREATE TABLE IF NOT EXISTS `bonus_script` (
+  `id` INT(11) NOT NULL AUTO_INCREMENT,
   `char_id` INT(11) UNSIGNED NOT NULL,
   `script` TEXT NOT NULL,
   `tick` BIGINT(20) NOT NULL DEFAULT '0',
   `flag` SMALLINT(5) UNSIGNED NOT NULL DEFAULT '0',
   `type` TINYINT(1) UNSIGNED NOT NULL DEFAULT '0',
-  `icon` SMALLINT(3) NOT NULL DEFAULT '-1'
+  `icon` SMALLINT(3) NOT NULL DEFAULT '-1',
+  PRIMARY KEY (`id`),
```

Comment:
The real primary key is char_id + type.
Defining it like this will also have the benefit that you dont need the seperate index for char_id.